app/src/test/java/de/westnordost/streetcomplete/data/osm/mapdata/MapDataCacheTest.kt

```diff
@@ -104,7 +105,7 @@ internal class MapDataCacheTest {
     }
 
     @Test fun `getWaysForNode gets filled inside bbox`() {
-        val node1 = node(1, LatLon(0.0, 0.0))
+        val node1 = node(1, LatLon(0.00001, 0.00001))
```

Comment:
I had to change `LatLon(0.0, 0.0)` to `LatLon(0.00001, 0.00001)`, because otherwise the node doesn't get put to SpatialCache.

This is because `listOf(LatLon(0.0, 0.0), LatLon(0.0002, 0.0002)).enclosingBoundingBox().enclosingTilesRect(16)` gives `TilesRect(left=32768, top=32767, right=32768, bottom=32767)`
But `LatLon(0.0, 0.0).position.enclosingTilePos(16)` is `TilePos(x=32768, y=32768)`, i.e. a tile that is *not* in above `TilesRect`

Reply:
This causes issues for other tests too.

Maybe https://github.com/streetcomplete/StreetComplete/blob/947fc8db1a53ef4d0949b6dbc5d2420596c6db91/app/src/main/java/de/westnordost/streetcomplete/util/SpatialCache.kt#L99-L104 needs to be switched back to use bbox. Or is the difference in above post actually a bug?

Reply:
Huh? What does `entries@for` even mean? I never encountered this syntax in Kotlin before. (By the way, for things that are basically if statements and not expressions, I very much prefer `if`, i.e. 
```
val tile = byTile[item.getTilePos()
if (tile != null) {
   ...
```

Regarding the question asked, not sure if I understand. Is the issue that the bounds of a bounding box are defined to be inclusive while the bounds of a tile are defined to be only inclusive on one side and exclusive on the other?

E.g. bbox(0,0,1,1) means p(0,0) is inside and p(1,1) is also inside but a theoretical tile covering that bbox would only have p(0,0) inside and p(0.99999999999999999,0.99999999999999) inside?

Reply:
> What does entries@for even mean?

Looks like I missed it when removing the loop over tiles inside the entries loop. It was used for `continue@entries`, i.e. continue in the outer loop.

The issue is:
```kotlin
val tilesRect = listOf(LatLon(0.0, 0.0), LatLon(0.0002, 0.0002)).enclosingBoundingBox().enclosingTilesRect(16)
val tilePos = LatLon(0.0, 0.0).enclosingTilePos(16)
assertTrue(tilesRect.contains(tilePos.toTilesRect()))
```
fails, and I don't understand whether it's ok that a `TilesRect` containing `LatLon(0.0, 0.0)` does not contain `LatLon(0.0, 0.0).enclosingTilePos(16).toTilesRect()`

Reply:
Yeah, this sounds suspect. Looks like a bug